lib/rouge/lexers/ecl.rb

```diff
@@ -118,7 +118,10 @@ def self.typed
         rule %r([\[\]{}();,\&,\.,\%]), Punctuation
 
         rule %r(\b(?i:(beginc\+\+.*?endc\+\+)))m, Str::Single
-        rule %r(\b(?i:(embed.*?endembed)))m, Str::Single
+        rule %r(\b(embed)([(])(.*?)([)])(.*?)(endembed))im do |m|
+          groups Keyword, Punctuation, Name::Other, Punctuation, Name::Other, Keyword
```

Comment:
For the first version I don't we should attempt to colorize anything within the `embed` / `endembed` sections (the content of these sections are other languages: Python, c++, Java, JavaScript, R...) and are only used in "rare" situations.

Reply:
@GordonSmith I don't mind either way. The rule was in the version of the lexer I edited but please feel free to remove and push.